src/com/facebook/buck/jvm/kotlin/KotlinLibraryDescription.java

```diff
@@ -193,8 +193,65 @@ public void findDepsForTargetFromConstructorArgs(
   }
 
   public interface CoreArg extends JavaLibraryDescription.CoreArg {
+    /**
+     * @deprecated use {@link #getFreeCompilerArgs()}.
+     */
+    @Deprecated
     ImmutableList<String> getExtraKotlincArguments();
 
+    ImmutableList<String> getFreeCompilerArgs();
+
+    @Value.Default
+    default boolean getAllWarningsAsErrors() {
+      return false;
+    }
+
+    @Value.Default
+    default boolean getSuppressWarnings() {
+      return false;
+    }
+
+    @Value.Default
+    default boolean getVerbose() {
+      return false;
+    }
+
+    @Value.Default
+    default boolean getIncludeRuntime() {
+      return false;
+    }
+
+    @Value.Default
+    default String getJvmTarget() {
+      return "1.6";
+    }
+
+    Optional<String> getJdkHome();
+
+    @Value.Default
+    default boolean getNoJdk() {
+      return false;
+    }
+
+    @Value.Default
+    default boolean getNoStdlib() {
+      return true;
+    }
+
+    @Value.Default
+    default boolean getNoReflect() {
+      return true;
+    }
+
+    @Value.Default
+    default boolean getJavaParameters() {
+      return false;
+    }
```

Comment:
also add a section in https://github.com/facebook/buck/blob/master/docs/rule/kotlin_library.soy describing these parameters.